Version command crash with Podman platform

```
$ export SKUPPER_PLATFORM=podman

$ skupper version
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x19bbfbe]

goroutine 1 [running]:
github.com/skupperproject/skupper/internal/cmd/skupper/version/nonkube.(*CmdVersion).NewClient(0xc00047c100, 0xc00047e008, {0x8?, 0xbee5f2?, 0xc0001a3e00?})
	/home/circleci/project/internal/cmd/skupper/version/nonkube/version.go:38 +0x5e
github.com/skupperproject/skupper/internal/cmd/skupper/version.CmdVersionFactory.ConfigureCobraCommand.func1(0xc00047e008, {0x3228020, 0x0, 0x0})
	/home/circleci/project/internal/cmd/skupper/common/command.go:50 +0x243
github.com/spf13/cobra.(*Command).execute(0xc00047e008, {0x3228020, 0x0, 0x0})
	/home/circleci/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:970 +0xa44
github.com/spf13/cobra.(*Command).ExecuteC(0x31b0300)
	/home/circleci/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(0x312f358?)
	/home/circleci/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041 +0x13
main.main()
	/home/circleci/project/cmd/skupper/main.go:15 +0x18
-> 2
```

Main (fb05630a)

--- a/internal/cmd/skupper/version/nonkube/version.go
+++ b/internal/cmd/skupper/version/nonkube/version.go
@@ -3,13 +3,14 @@ package nonkube
 import (
 	"errors"
 	"fmt"
+	internalclient "github.com/skupperproject/skupper/internal/nonkube/client/compat"
+	"github.com/skupperproject/skupper/pkg/nonkube/api"
 	"os"
 	"text/tabwriter"
 
 	"github.com/skupperproject/skupper/internal/cmd/skupper/common"
 	"github.com/skupperproject/skupper/internal/cmd/skupper/common/utils"
 	"github.com/skupperproject/skupper/internal/images"
-	"github.com/skupperproject/skupper/internal/kube/client"
 	"github.com/skupperproject/skupper/internal/utils/configs"
 	"github.com/skupperproject/skupper/internal/utils/validator"
 	skupperv2alpha1 "github.com/skupperproject/skupper/pkg/generated/client/clientset/versioned/typed/skupper/v2alpha1"
@@ -35,10 +36,8 @@ func NewCmdVersion() *CmdVersion {
 }
 
 func (cmd *CmdVersion) NewClient(cobraCommand *cobra.Command, args []string) {
-	cli, err := client.NewClient(cobraCommand.Flag("namespace").Value.String(), cobraCommand.Flag("context").Value.String(), cobraCommand.Flag("kubeconfig").Value.String())
-
-	if err == nil {
-		cmd.namespace = cli.Namespace
+	if cmd.CobraCmd != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace) != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String() != "" {
+		cmd.namespace = cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String()
 	}
 }
 
@@ -55,14 +54,35 @@ func (cmd *CmdVersion) ValidateInput(args []string) error {
 		}
 	}
 
+	_, err := os.Stat(api.GetHostNamespaceHome(cmd.namespace))
+	if err != nil {
+		validationErrors = append(validationErrors, fmt.Errorf("there is no definition for namespace %q", cmd.namespace))
+	}
+
 	return errors.Join(validationErrors...)
 }
 
 func (cmd *CmdVersion) InputToOptions() {
+
+	mapRunningPods := make(map[string]string)
+
+	if cmd.namespace == "" {
+		cmd.namespace = "default"
+	}
+
+	cli, err := internalclient.NewCompatClient(os.Getenv("CONTAINER_ENDPOINT"), "")
+
+	if err == nil {
+		containerName := cmd.namespace + "-skupper-router"
+		if container, err := cli.ContainerInspect(containerName); err == nil {
+			mapRunningPods[container.Name] = container.Image
+		}
+	}
+
 	if cmd.output != "" {
-		cmd.manifest = configs.ManifestManager{Components: images.NonKubeComponents, EnableSHA: true}
+		cmd.manifest = configs.ManifestManager{Components: images.NonKubeComponents, EnableSHA: true, RunningPods: mapRunningPods}
 	} else {
-		cmd.manifest = configs.ManifestManager{Components: images.NonKubeComponents, EnableSHA: false}
+		cmd.manifest = configs.ManifestManager{Components: images.NonKubeComponents, EnableSHA: false, RunningPods: mapRunningPods}
 	}
 
 }

--- a/internal/cmd/skupper/version/nonkube/version_test.go
+++ b/internal/cmd/skupper/version/nonkube/version_test.go
@@ -19,6 +19,7 @@ func TestCmdVersion_ValidateInput(t *testing.T) {
 		name           string
 		args           []string
 		flags          common.CommandVersionFlags
+		namespace      string
 		k8sObjects     []runtime.Object
 		skupperObjects []runtime.Object
 		expectedError  string
@@ -26,27 +27,36 @@ func TestCmdVersion_ValidateInput(t *testing.T) {
 
 	testTable := []test{
 		{
-			name:          "bad output",
+			name:          "incorrect output type",
 			args:          []string{""},
+			namespace:     "test",
 			flags:         common.CommandVersionFlags{Output: "not-supported"},
 			expectedError: "output type is not valid: value not-supported not allowed. It should be one of this options: [json yaml]",
 		},
 		{
-			name:          "good output",
+			name:          "valid output type",
+			namespace:     "test",
 			flags:         common.CommandVersionFlags{Output: "json"},
 			expectedError: "",
 		},
 		{
-			name:          "ok no output",
+			name:          "unspecified output type",
+			namespace:     "test",
 			flags:         common.CommandVersionFlags{},
 			expectedError: "",
+		},
+		{
+			name:          "unknown namespace",
+			namespace:     "unknown",
+			flags:         common.CommandVersionFlags{},
+			expectedError: "there is no definition for namespace \"unknown\"",
 		},
 	}
 
 	for _, test := range testTable {
 		t.Run(test.name, func(t *testing.T) {
 
-			cmd, err := newCmdVersionWithMocks("test", test.k8sObjects, test.skupperObjects, "")
+			cmd, err := newCmdVersionWithMocks(test.namespace)
 			assert.Assert(t, err)
 
 			cmd.Flags = &test.flags
@@ -67,12 +77,12 @@ func TestCmdVersion_InputToOptions(t *testing.T) {
 
 	testTable := []test{
 		{
-			name:     "good json",
+			name:     "output type selected is json",
 			output:   "json",
 			expected: true,
 		},
 		{
-			name:     "good defaul",
+			name:     "output type selected is the default one",
 			expected: false,
 		},
 	}
@@ -80,7 +90,7 @@ func TestCmdVersion_InputToOptions(t *testing.T) {
 	for _, test := range testTable {
 		t.Run(test.name, func(t *testing.T) {
 
-			cmd, err := newCmdVersionWithMocks("test", test.k8sObjects, test.skupperObjects, "")
+			cmd, err := newCmdVersionWithMocks("test")
 			assert.Assert(t, err)
 
 			cmd.output = test.output
@@ -119,7 +129,7 @@ func TestCmdVersion_Run(t *testing.T) {
 	}
 
 	for _, test := range testTable {
-		cmd, err := newCmdVersionWithMocks("test", test.k8sObjects, test.skupperObjects, test.skupperErrorMessage)
+		cmd, err := newCmdVersionWithMocks("test")
 		assert.Assert(t, err)
 		cmd.Flags = &test.flags
 		cmd.output = cmd.Flags.Output
@@ -140,7 +150,7 @@ func TestCmdVersion_Run(t *testing.T) {
 
 // --- helper methods
 
-func newCmdVersionWithMocks(namespace string, k8sObjects []runtime.Object, skupperObjects []runtime.Object, fakeSkupperError string) (*CmdVersion, error) {
+func newCmdVersionWithMocks(namespace string) (*CmdVersion, error) {
 
 	cmdVersion := &CmdVersion{
 		namespace: namespace,

